Update: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type.md (#4513)
umbraco/UmbracoDocs: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type.md

        this.Name = "Long Answer with Limit";
    }

>     public override IEnumerable<string> ValidateField(Form form, Field field, IEnumerable<object> postedValues, HttpContext context, IPlaceholderParsingService placeholderParsingService, List<string> errors)

Reviewer: Thanks, this got omitted from being updated from the Forms 8 version by the looks of it, so good to have it correct now.

    }
}
```

> In addition to the above, you must also register the extended field type within a composer and also create the backoffice field type view.

Reviewer: ```suggestion
As discussed in the previous section, you must also register the extended field type within a composer.  You also need to create the the backoffice field type view.
```

This is partly covered in the section above, so is slightly repeated here.  No harm in that though, so suggest we keep but relate it to the previous copy.

**Composer:**
```csharp
public class UmbracoFormsCustomProvidersComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
    {
        builder.FormsFields().Add<TextareaWithCount>();
    }
}
```

**Backoffice View:**

> Add a new HTML file as per the name of the field class (e.g. ```textareawithcount.html```) to ```\wwwroot\App_Plugins\umbracoforms\Backoffice\Common\FieldTypes\``` within your project. For this example we can just copy the original ```textarea.html``` file used by the standard 'Long Answer' field.

Reviewer 2: ```suggestion
Add a new HTML file as per the name of the field class (e.g. `textareawithcount.html`) to `\wwwroot\App_Plugins\umbracoforms\Backoffice\Common\FieldTypes\` within your project. For this example, we can just copy the original `textarea.html` file used by the standard 'Long Answer' field.
```

> If you do not have the original ```\wwwroot\App_Plugins\umbracoforms\``` folders/files (which are otherwise embedded within a RCL of the package) then you can download these from the 'Package Files' section of the [Umbraco Forms Package](https://our.umbraco.com/packages/developer-tools/umbraco-forms/#tab-package) typically named as ```UmbracoForms.Files.{version}.zip```. 

Reviewer 2: ```suggestion
If you do not have the original `\wwwroot\App_Plugins\umbracoforms\` folders/files they can be downloaded. This can be done from the 'Package Files' section of the [Umbraco Forms Package](https://our.umbraco.com/packages/developer-tools/umbraco-forms/#tab-package). It will typically be named `UmbracoForms.Files.{version}.zip`.
```

> If you do not have the original ```\wwwroot\App_Plugins\umbracoforms\``` folders/files (which are otherwise embedded within a RCL of the package) then you can download these from the 'Package Files' section of the [Umbraco Forms Package](https://our.umbraco.com/packages/developer-tools/umbraco-forms/#tab-package) typically named as ```UmbracoForms.Files.{version}.zip```. 

Reviewer: We have details about this already [here](https://our.umbraco.com/documentation/Add-ons/UmbracoForms/Developer/Themes/#creating-a-theme), but there's no harm in repeating it.  As we aren't linking to a specific zip file, we aren't adding to the maintenance costs of updated documentation when new versions are released.

Reviewer 2: Can I suggest that we instead of having it twice, add a link from this article to the "Themes" article? And maybe associate the link with a few lines of text.
